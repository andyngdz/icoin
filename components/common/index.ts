export { ICoinLogo } from 'components/common/Logo'
export { Route } from 'components/common/Route'
