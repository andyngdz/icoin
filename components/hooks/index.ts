export * from './useChartData'
export * from './useGlobal'
export * from './useLivePrice'
export * from './useLivePrices'
export * from './usePagination'
export * from './useTheme'
export * from './useTime'
