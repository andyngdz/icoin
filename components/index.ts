export * from './asset-detail/AssetDetail'
export * from './asset-detail/AssetDetailContent'
export * from './asset-detail/AssetHighLow'
export * from './asset-detail/AssetSummary'
export * from './assets/AssetItem'
export * from './assets/AssetNameAndDescription'
export * from './assets/Assets'
export * from './assets/AssetsTable'
export * from './common/container-wrapper/ContainerWrapper'
export * from './common/logo/Logo'
export * from './common/pagination/Pagination'
export * from './common/paper-wrapper/PaperWrapper'
export * from './common/route/Route'
export * from './common/value-color/ValueColor'
export * from './disclaimer/Disclaimer'
export * from './exchanges/ExchangeContent'
export * from './exchanges/ExchangeItem'
export * from './exchanges/ExchangesAll'
export * from './exchanges/ExchangesAsset'
export * from './exchanges/ExchangeStatus'
export * from './header/Header'
export * from './header/Menu'
export * from './hooks'
export * from './line-chart/LineChart'
export * from './live-prices/LivePrices'
export * from './search/Search'
export * from './symbol-icon/SymbolIcon'
export * from './time-selection/TimeSelection'
export * from './top-bar/TopBar'
export * from './top-bar/TopBarContent'
