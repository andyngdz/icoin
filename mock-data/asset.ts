const asset = {
  changePercent24Hr: 3.1752942454749208,
  explorer: 'https://blockchain.info/',
  id: 'bitcoin',
  marketCapUsd: 649529734588.947600205682695,
  maxSupply: 21000000.0,
  name: 'Bitcoin',
  priceUsd: 34888.9719149353737679,
  rank: 1,
  supply: 18617050.0,
  symbol: 'BTC',
  volumeUsd24Hr: 14661390048.4857838209011476,
  vwap24Hr: 34081.3126994825452706
}

export { asset }
