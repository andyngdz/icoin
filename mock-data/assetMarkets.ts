const assetMarkets = {
  pageInfo: { hasNextPage: true, __typename: 'PageInfo' },
  edges: [
    {
      node: {
        id: '3a6e1109-9f90-4659-8985-5c8b9e36328b',
        exchangeId: 'binance',
        exchangeName: 'Binance',
        baseSymbol: 'BTC',
        quoteSymbol: 'USDT',
        volumeUsd24Hr: 3822582086.935666004007041,
        priceUsd: 47483.608071927724242,
        percentVolume: 18.2682261132248922,
        updatedAt: '2021-02-15T08:48:27.353Z',
        __typename: 'AssetMarket'
      },
      __typename: 'AssetMarketEdge'
    },
    {
      node: {
        id: '04a633d2-d43f-4b73-9e0c-22fc9195af8a',
        exchangeId: 'bit-z',
        exchangeName: 'Bit-Z',
        baseSymbol: 'BTC',
        quoteSymbol: 'USDT',
        volumeUsd24Hr: 1543652315.1777949383004728,
        priceUsd: 47489.0307680033636913,
        percentVolume: 7.3771573487587625,
        updatedAt: '2021-02-15T08:48:12.995Z',
        __typename: 'AssetMarket'
      },
      __typename: 'AssetMarketEdge'
    },
    {
      node: {
        id: '00ae5690-a93a-41aa-917f-6fc4b6e03b56',
        exchangeId: 'huobi',
        exchangeName: 'Huobi',
        baseSymbol: 'BTC',
        quoteSymbol: 'USDT',
        volumeUsd24Hr: 1466373888.025307072992641,
        priceUsd: 47504.9586871037659115,
        percentVolume: 7.0078415960059594,
        updatedAt: '2021-02-15T08:35:29.911Z',
        __typename: 'AssetMarket'
      },
      __typename: 'AssetMarketEdge'
    }
  ],
  __typename: 'AssetMarketConnection'
}

export { assetMarkets }
