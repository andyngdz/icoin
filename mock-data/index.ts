export * from './asset'
export * from './assetMarkets'
export * from './assetNode'
export * from './assets'
export * from './assetsString'
export * from './exchanges'
export * from './global-data'
export * from './histories'
export * from './historyString'
export * from './node'
