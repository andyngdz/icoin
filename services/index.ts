export * from './paginate'
export * from './chart'
export * from './format'
export * from './api'
export * from './cdn'
export * from './test-utils'
export * from './price-direction'
export * from './price'
export * from './routes'
