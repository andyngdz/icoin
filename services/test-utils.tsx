import { Provider } from 'react-redux'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@material-ui/core'
import React from 'react'
import Store from 'states'
import Theme from 'styles'

const Providers = ({ children }) => {
  return (
    <Provider store={Store}>
      <ThemeProvider theme={Theme}>{children}</ThemeProvider>
    </Provider>
  )
}

const customRender = (ui, options = {}) =>
  render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'

export { customRender as render }
