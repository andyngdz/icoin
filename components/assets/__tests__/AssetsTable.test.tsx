import { act, customRender, RenderResult } from 'services/test-utils'
import { AssetsTable } from 'components'
import { assets } from 'mock-data'

describe('AssetsTable', () => {
  it('Should render assets table correctly', async () => {
    const { data } = assets
    let renderResult: RenderResult

    await act(async () => {
      renderResult = customRender(<AssetsTable data={data} />)
    })

    expect(renderResult).toMatchSnapshot()
  })
})
