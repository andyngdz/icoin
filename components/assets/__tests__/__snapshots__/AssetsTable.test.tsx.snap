// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetsTable Should render assets table correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="MuiTable-root"
      >
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiTableSortLabel-root MuiTableSortLabel-active"
                role="button"
                tabindex="0"
              >
                Rank
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiTableSortLabel-root"
                role="button"
                tabindex="0"
              >
                Name and Description
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiTableSortLabel-root"
                role="button"
                tabindex="0"
              >
                Market Cap
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiTableSortLabel-root"
                role="button"
                tabindex="0"
              >
                Price
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiTableSortLabel-root"
                role="button"
                tabindex="0"
              >
                Volume
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiTableSortLabel-root"
                role="button"
                tabindex="0"
              >
                Change (24h)
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiTableSortLabel-root"
                role="button"
                tabindex="0"
              >
                Circularing Supply
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root AssetItem-asset-1"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              1
              .
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              <div
                class="AssetNameAndDescription-name-4"
              >
                <div
                  class="AssetNameAndDescription-symbol-5"
                />
                <div>
                  <p
                    class="MuiTypography-root MuiTypography-button"
                  >
                    Bitcoin
                  </p>
                  <span
                    class="MuiTypography-root MuiTypography-caption"
                  >
                    BTC
                  </span>
                </div>
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              $647,158,791,754
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              $34,761.62
               
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              $14,567,120,537
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              <span
                class="MuiTypography-root ValueColor-up-7 MuiTypography-body2"
              >
                3.15%
              </span>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              18,617,050
            </td>
          </tr>
          <tr
            class="MuiTableRow-root AssetItem-asset-1"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              2
              .
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              <div
                class="AssetNameAndDescription-name-4"
              >
                <div
                  class="AssetNameAndDescription-symbol-5"
                />
                <div>
                  <p
                    class="MuiTypography-root MuiTypography-button"
                  >
                    Ethereum
                  </p>
                  <span
                    class="MuiTypography-root MuiTypography-caption"
                  >
                    ETH
                  </span>
                </div>
              </div>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              $163,024,735,364
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              $1,423.75
               
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              $10,514,268,643
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              <span
                class="MuiTypography-root ValueColor-up-7 MuiTypography-body2"
              >
                7.16%
              </span>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              114,503,686.687
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="MuiTable-root"
    >
      <thead
        class="MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiTableSortLabel-root MuiTableSortLabel-active"
              role="button"
              tabindex="0"
            >
              Rank
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                />
              </svg>
            </span>
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head"
            scope="col"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiTableSortLabel-root"
              role="button"
              tabindex="0"
            >
              Name and Description
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                />
              </svg>
            </span>
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
            scope="col"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiTableSortLabel-root"
              role="button"
              tabindex="0"
            >
              Market Cap
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                />
              </svg>
            </span>
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
            scope="col"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiTableSortLabel-root"
              role="button"
              tabindex="0"
            >
              Price
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                />
              </svg>
            </span>
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
            scope="col"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiTableSortLabel-root"
              role="button"
              tabindex="0"
            >
              Volume
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                />
              </svg>
            </span>
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
            scope="col"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiTableSortLabel-root"
              role="button"
              tabindex="0"
            >
              Change (24h)
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                />
              </svg>
            </span>
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
            scope="col"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiTableSortLabel-root"
              role="button"
              tabindex="0"
            >
              Circularing Supply
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                />
              </svg>
            </span>
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root AssetItem-asset-1"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            1
            .
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            <div
              class="AssetNameAndDescription-name-4"
            >
              <div
                class="AssetNameAndDescription-symbol-5"
              />
              <div>
                <p
                  class="MuiTypography-root MuiTypography-button"
                >
                  Bitcoin
                </p>
                <span
                  class="MuiTypography-root MuiTypography-caption"
                >
                  BTC
                </span>
              </div>
            </div>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            $647,158,791,754
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            $34,761.62
             
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            $14,567,120,537
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            <span
              class="MuiTypography-root ValueColor-up-7 MuiTypography-body2"
            >
              3.15%
            </span>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            18,617,050
          </td>
        </tr>
        <tr
          class="MuiTableRow-root AssetItem-asset-1"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            2
            .
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body"
          >
            <div
              class="AssetNameAndDescription-name-4"
            >
              <div
                class="AssetNameAndDescription-symbol-5"
              />
              <div>
                <p
                  class="MuiTypography-root MuiTypography-button"
                >
                  Ethereum
                </p>
                <span
                  class="MuiTypography-root MuiTypography-caption"
                >
                  ETH
                </span>
              </div>
            </div>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            $163,024,735,364
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            $1,423.75
             
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            $10,514,268,643
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            <span
              class="MuiTypography-root ValueColor-up-7 MuiTypography-body2"
            >
              7.16%
            </span>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
          >
            114,503,686.687
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
